Lib/test/test_type_params.py

```diff
@@ -1240,6 +1243,32 @@ def test_pickling_classes(self):
                     # but class check is good enough:
                     self.assertIsInstance(pickle.loads(pickled), real_class)
 
+    def test_pickling_anonymous_typeparams(self):
+        # see gh-129250
+        thing = func5()
```

Comment:
Can you also test a generic method? This would make sure we correctly use the qualname rather than just the name.